java/core/src/java/org/apache/orc/impl/BufferChunk.java

```diff
@@ -69,22 +69,8 @@ public final String toString() {
 
   @Override
   public DiskRange sliceAndShift(long offset, long end, long shiftBy) {
-    assert offset <= end && offset >= this.offset && end <= this.end;
-    assert offset + shiftBy >= 0;
-    ByteBuffer sliceBuf = chunk.slice();
-    int newPos = (int) (offset - this.offset);
-    int newLimit = newPos + (int) (end - offset);
-    try {
-      sliceBuf.position(newPos);
-      sliceBuf.limit(newLimit);
-    } catch (Throwable t) {
-      LOG.error("Failed to slice buffer chunk with range" + " [" + this.offset + ", " + this.end
-          + "), position: " + chunk.position() + " limit: " + chunk.limit() + ", "
-          + (chunk.isDirect() ? "direct" : "array") + "; to [" + offset + ", " + end + ") "
-          + t.getClass());
-      throw new RuntimeException(t);
-    }
-    return new BufferChunk(sliceBuf, offset + shiftBy);
+    // Method has no code references
+    throw new UnsupportedOperationException();
```

Comment:
If this method is not used, we need to to in an independent PR.
Please revert this change from this PR.

Reply:
Of course, we should evaluate the removal PR first.
For that PR, we need to validate with ORC/Iceberg/Hive at least.
Although there exists many downstream, currently only the above three are compatible with ORC 1.6.x.
We don't want to lose them again.

Reply:
I have gone through the change again carefully. This is not required for the current implementation, so I will revert this change here and we can take this up independently if needed